views/apartments/index.erb

```diff
@@ -1,5 +1,14 @@
+<h2>Available Apartments</h2>
+
 <ul>
-  <li><a href='/apartments/1'>Apartment 1</a></li>
-  <li><a href='/apartments/2'>Apartment 2</a></li>
-  <li><a href='/apartments/3'>Apartment 3</a></li>
+  <% @apartments.each do |apartment| %>
+    <li><p>
+      <a href="/apartments/<%= apartment.id %>"><%= apartment.address %> </a> --
+      <a href="/apartments/<%= apartment.id %>/edit">edit</a> --
+      <a href="/apartments/<%= apartment.id %>/delete">delete</a></p>
+    </li>
+  <% end %>
```

Comment:
Great job adding all the relevant links.

One small note... the delete link won't work, since links always make GET requests, but deleting something needs to make a DELETE request according to your controller file.

The correct way is to write your controller as you have (using a DELETE request), but to make all delete 'links' really HTML forms behind the scenes. See Tunr for an example.